lock-api-objects/src/main/java/com/palantir/lock/watch/SpanningCommitUpdate.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.watch;
+
+import com.palantir.common.annotations.ImmutablesStyles.StagedBuilderStyle;
+import com.palantir.lock.watch.ImmutableSpanningCommitUpdate.TransactionCommitUpdateBuildStage;
+import org.immutables.value.Value;
+
+@Value.Immutable
+@StagedBuilderStyle
+public interface SpanningCommitUpdate {
+    CommitUpdate transactionCommitUpdate();
+
+    CommitUpdate spanningCommitUpdate();
```

Comment:
Staged to prevent crossing wires!

Reply:
I don't think staged is really necessary -- they are already not parameters, so it's pretty difficult to cross wires.
One option here is, if you go with the `MaybeSpanningLockDescriptor` approach, to just have a factory that takes a set of those and then constructs the correct thing -- they can then even actually be `@Value.Lazy` so you prevent some of the extra work for the case where you do not care about spanning update. (That last one reminds me too much of timelock partitioning so probably best not to)

Reply:
Eh, I like staged builders anyway so I'll stick with it (and no no no to `@Value.Lazy`!)